### PR TITLE
clear cookies in web view

### PR DESCRIPTION
**Pull request by ayushmeghwani**

### CODE REVIEW

**Comment by shahzaibj** [comment]
Why?

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java [hunk 1]
```diff
@@ -198,6 +198,10 @@ private void setUpWebView(@NonNull final View view,
         mWebView.getSettings().setJavaScriptEnabled(true);
         mWebView.requestFocus(View.FOCUS_DOWN);
 
+        // clear cookies from web view session before processing new login request
+        WebViewUtil.removeCookiesFromWebView(getContext());
+        WebViewUtil.removeSessionCookiesFromWebView(getContext());
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java:203]: The documentation for both of these methods state that they [cannot be called from the UI thread](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/ui/webview/WebViewUtil.java#L87)

Comment (iambmelt) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java:203]: (In general though, I do not think we should take this PR -- see my other comment)

**Review by iambmelt — CHANGES_REQUESTED** [review]
I am unclear about the purpose/intent of this change; at a glance, I don't believe this is the correct behavior:
- The cookie jar is scoped to _all_ of this applications `WebViews`, not just the instance used for authentication. As a result, performing interactive auth will eject any non-msft cookies set on the `WebView` which will lead to bugs/loss of session state for any sites visited by the user that may be unrelated to MSFT auth

File: common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java (hunk 1 above)
Comment (shahzaibj) [reply common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java:203]: +1 - I also don't know why are making this change. I think we want the cookies to be there. @ayushmeghwani can you explain the rationale behind this change?

**Comment by iambmelt** [comment]
Closing due to inactivity. Do not presently need this change

**Pull request closed without merging**